x-pack/plugins/observability_ai_assistant/server/service/client/index.ts

```diff
@@ -135,7 +135,42 @@ export class ObservabilityAIAssistantClient {
         })
     );
 
-    const functionsForOpenAI: ChatCompletionFunctions[] | undefined = functions;
+    // add recalled information to system message, so the LLM considers it more important
+
+    const recallMessages = messagesForOpenAI.filter((message) => message.name === 'recall');
+
+    const recalledDocuments: Map<string, { id: string; text: string }> = new Map();
+
+    recallMessages.forEach((message) => {
+      const entries = message.content
+        ? (JSON.parse(message.content) as Array<{ id: string; text: string }>)
+        : [];
+
+      const ids: string[] = [];
+
+      entries.forEach((entry) => {
+        const id = entry.id;
+        if (!recalledDocuments.has(id)) {
+          recalledDocuments.set(id, entry);
+        }
+        ids.push(id);
+      });
+
+      message.content = `The following documents, present in the system message, were recalled: ${ids.join(
```

Comment:
If the user uses the assistant in a different language than English, or the KB is filled with entries that are not in English, I wonder when we suddenly add in English like this (`The following documents...`) impacts performance in some way.

Reply:
Yeah i see what you mean. it's a "problem" we have all over the place, function responses, system message etc are all in English. I assume the LLM handles this for us, unless proven otherwise.